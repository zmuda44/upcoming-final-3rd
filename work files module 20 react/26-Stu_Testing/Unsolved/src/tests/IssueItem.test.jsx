import pretty from 'pretty';
import { render } from '@testing-library/react';
// TODO: Import the IssueItem component here

const issue = {
  url: 'https://api.github.com/repos/microsoft/vscode/issues/68',
  html_url: 'https://github.com/microsoft/vscode/issues/68',
  id: 117635943,
  node_id: 'MDU6SXNzdWUxMTc2MzU5NDM=',
  number: 68,
  title: 'Git: Support git history in VSCode',
};

describe('IssueItem', () => {
  it('should contain the expected text', () => {
    // TODO: Add a render code block that checks to see if the component renders properly
    // HINT: The render method needs a target element
    render();

    expect(document.querySelector('.item').textContent).toBe(
      'Git: Support git history in VSCode'
    );
  });

  // TODO: render the IssueItem component and format the rendered HTML with the pretty package before saving it as an snapshot.
  it('should match snapshot', () => {});
});
