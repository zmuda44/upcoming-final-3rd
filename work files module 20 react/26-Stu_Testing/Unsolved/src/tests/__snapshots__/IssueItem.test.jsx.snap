// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`IssueItem > should match snapshot 1`] = `
"<i class=\\"large github middle aligned icon\\"></i>
<div class=\\"content\\"><a class=\\"header\\" href=\\"https://github.com/microsoft/vscode/issues/68\\" target=\\"_blank\\" rel=\\"noreferrer\\">Git: Support git history in VSCode</a>
  <div class=\\"description\\"></div>
</div>"
`;
